bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1641,7 +1639,7 @@ OperationResult PreProcessor::launchReqPreProcessing(const PreProcessRequestMsgS
   LOG_DEBUG(
       logger(),
       "Pre-execution operation successfully completed" << KVLOG(cid, reqSeqNum, clientId, reqOffsetInBatch, blockId));
```

Comment:
It is not correct to copy the block id to the (preProcessResultBuffer + request.outActualReplySize) address in case of error. Please set resultLen to request.outActualReplySize in SUCCESSFUL case and copy the block to (preProcessResultBuffer + resultLen) address.